### PR TITLE
[Refactor] only use compute node to create cloudnative table

### PR DESCRIPTION
**Pull request by abc982627271**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/lake/backup/LakeRestoreJob.java [hunk 1]
```diff
@@ -125,7 +125,7 @@ protected void prepareAndSendSnapshotTasks(Database db) {
                 Partition part = tbl.getPartition(idChain.getPartId());
                 MaterializedIndex index = part.getIndex(idChain.getIdxId());
                 tablet = (LakeTablet) index.getTablet(idChain.getTabletId());
-                Backend backend = GlobalStateMgr.getCurrentSystemInfo().getBackend(tablet.getPrimaryBackendId());
+                Backend backend = GlobalStateMgr.getCurrentSystemInfo().getBackend(tablet.getPrimaryComputeNodeId());
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/lake/backup/LakeRestoreJob.java:128]: There is no guarantee the return value is an instance of `Backend`, maybe an instance of `ComputeNode`

File: be/src/agent/agent_server.cpp [hunk 2]
```diff
@@ -285,6 +284,9 @@ AgentServer::Impl::~Impl() {}
 // TODO(lingbin): each task in the batch may have it own status or FE must check and
 // resend request when something is wrong(BE may need some logic to guarantee idempotence.
 void AgentServer::Impl::submit_tasks(TAgentResult& agent_result, const std::vector<TAgentTaskRequest>& tasks) {
+    // for debug
+    LOG(INFO) << "enter submit_tasks";
```
Comment (sduzh) [inline be/src/agent/agent_server.cpp:288]: Delete or use VLOG

File: be/src/agent/agent_task.cpp [hunk 3]
```diff
@@ -209,6 +213,9 @@ void run_create_tablet_task(const std::shared_ptr<CreateTabletAgentTaskRequest>&
         tablet_info.__set_path_hash(tablet->data_dir()->path_hash());
     }
 
+    // for debug
+    LOG(INFO) << "ready to finish task to fe";
```
Comment (sduzh) [inline be/src/agent/agent_task.cpp:217]: ditto

File: fe/fe-core/src/main/java/com/starrocks/system/HeartbeatMgr.java [hunk 4]
```diff
@@ -239,6 +239,9 @@ private boolean handleHbResponse(HeartbeatResponse response, boolean isReplay) {
                         if (RunMode.allowCreateLakeTable() && !isReplay) {
                             // addWorker
                             int starletPort = computeNode.getStarletPort();
+                            // for debug
+                            LOG.info("will add worker, starletPort is {}", starletPort);
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/system/HeartbeatMgr.java:243]: Delete or VLOG

File: fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java [hunk 5]
```diff
@@ -1082,6 +1115,10 @@ private long getClusterAvailableCapacityB() {
     }
 
     public void checkClusterCapacity() throws DdlException {
+        if (RunMode.getCurrentRunMode() != RunMode.SHARED_DATA) {
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java:1118]: Better submit this patch in a separate PR.

File: fe/fe-core/src/main/java/com/starrocks/server/RunMode.java [hunk 6]
```diff
@@ -141,7 +141,7 @@ public boolean isAllowCreateLakeTable() {
 
         @Override
         public short getDefaultReplicationNum() {
-            return 3;
+            return 1;
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/server/RunMode.java:144]:
```suggestion
            return 3;
```

File: fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java [hunk 7]
```diff
@@ -388,13 +388,18 @@ private static void checkReplicationNum(short replicationNum) throws AnalysisExc
...
+                GlobalStateMgr.getCurrentSystemInfo().getAvailableBackendIds();
+
         if (RunMode.getCurrentRunMode() == RunMode.SHARED_DATA) {
             if (RunMode.defaultReplicationNum() > backendIds.size()) {
-                throw new AnalysisException("Number of available BE nodes is " + backendIds.size()
+                throw new AnalysisException("Number of available CN nodes is " + backendIds.size()
                         + ", less than " + RunMode.defaultReplicationNum());
             }
         } else {
+            GlobalStateMgr.getCurrentSystemInfo().getAvailableBackendIds();
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java:402]: Unused function call

File: fe/fe-core/src/main/java/com/starrocks/leader/LeaderImpl.java [hunk 8]
```diff
@@ -182,17 +183,24 @@ public TMasterResult finishTask(TFinishTaskRequest request) {
...
-            tStatus.setStatus_code(TStatusCode.CANCELLED);
-            List<String> errorMsgs = new ArrayList<>();
-            errorMsgs.add("backend not exist.");
-            tStatus.setError_msgs(errorMsgs);
-            LOG.warn("backend does not found. host: {}, be port: {}. task: {}", host, bePort, request.toString());
-            return result;
+        long backendId = -1L;
+        if (Config.only_use_compute_node) {
+            ComputeNode cn = GlobalStateMgr.getCurrentSystemInfo().getComputeNodeWithBePort(host, bePort);
+            backendId = cn.getId();
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/leader/LeaderImpl.java:189]: May throw NPE

File: fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java [hunk 9]
```diff
@@ -582,6 +593,13 @@ public int getTotalBackendNumber() {
         return idToBackendRef.size();
     }
 
+    public int getAliveComputeNodeNumber() {
+        if (Config.only_use_compute_node) {
+            return getComputeNodeIds(true).size();
+        }
+        return getBackendIds(true).size();
+    }
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java:601]: Misleading function name when `Config.only_use_compute_node` is false.

**Review by sduzh — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java [hunk 10]
```diff
@@ -1082,6 +1111,10 @@ private long getClusterAvailableCapacityB() {
     }
 
     public void checkClusterCapacity() throws DdlException {
+        if (RunMode.getCurrentRunMode() != RunMode.SHARED_DATA) {
+            return;
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java:1115]: return if run mode == shared data

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 11]
```diff
@@ -1658,19 +1658,20 @@ void buildPartitions(Database db, OlapTable table, List<Partition> partitions) t
...
             LOG.info("creating {} partitions of table {} concurrently", partitions.size(), table.getName());
-            buildPartitionsConcurrently(db.getId(), table, partitions, numReplicas, numAliveBackends);
-        } else if (numAliveBackends > 0) {
-            buildPartitionsSequentially(db.getId(), table, partitions, numReplicas, numAliveBackends);
+            buildPartitionsConcurrently(db.getId(), table, partitions, numReplicas, numAliveCNs);
+        } else if (numAliveCNs > 0) {
+            buildPartitionsSequentially(db.getId(), table, partitions, numReplicas, numAliveCNs);
         } else {
-            throw new DdlException("no alive backend");
+            throw new DdlException("no alive compute nodes");
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:1674]: Is this error also appropriate for local tables?

**Review by wyb — DISMISSED** [review]

**Review by sduzh — DISMISSED** [review]

**Review by wyb — APPROVED** [review]

**Review by sduzh — APPROVED** [review]
